### PR TITLE
Update nuget client

### LINKED ISSUE #224: nuspec format support errors
Hi! Thanks for reporting this feature/bug/question! 
 
Please keep / fill in the relevant info from this template so that we can help you as best as possible.

**Type** (choose one):

- Bug
snag_8-27-2017_17-47-30

**Installed from**:  chocolatey.org

In case of a BUG:

snag_8-27-2017_17-46-21

This has been coming up almost continually with all new packages lately, so I'm not sure if folks are just starting to utilize newer metadata elements that just have not been added to this explorer's parser code, or ? but if I edit the xml and remove the elements it complains about, then it functions just fine.

### PR DESCRIPTION
**Pull request by onovotny**
Branch to move to the newer Nuget.Client. This builds but needs a thorough review of all editors for type conversion. Also needs review of all XAML bindings due to internal OM changes.

Fixes #224 & #230 

This will be the first of several PR's implementing features based on the new client libraries. Initially, this PR is just to preserve existing functionality using the new libraries.

### CODE REVIEW

**Comment by 304NotModified** [comment]
also fixes https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/issues/230

File: Core/Authoring/ManifestDependency.cs [hunk 1]
```diff
@@ -17,5 +17,8 @@ public class ManifestDependency
 
         [XmlAttribute("exclude")]
         public string Exclude { get; set; }
+
+        [XmlAttribute("include")]
+        public string Include { get; set; }
```
Comment (304NotModified) [inline Core/Authoring/ManifestDependency.cs]: guess we are missing "include in the GUI", create a new issue for that?

image

**Comment by 304NotModified** [comment]
confirm fixes #230

File: Core/Authoring/PackageBuilder.cs [hunk 2]
```diff
@@ -257,11 +258,11 @@ internal static void ValidateReferenceAssemblies(IEnumerable<IPackageFile> files
 
         private static bool IsPrereleaseDependency(PackageDependency dependency)
         {
-            var versionSpec = dependency.VersionSpec;
+            var versionSpec = dependency.VersionRange;
```
Comment (304NotModified) [inline Core/Authoring/PackageBuilder.cs]: guess we should deduplicate this `IsPrereleaseDependency` with `PrereleasePackageDependencyRule.IsPrereleaseDependency` (or in the future), do you agree?

File: Core/Extensions/CredentialProviderExtensions.cs [hunk 3]
```diff
@@ -1,48 +0,0 @@
-﻿using System;
-using System.Net;
-
-namespace NuGet
-{
-    internal static class CredentialProviderExtensions
```
Comment (304NotModified) [inline Core/Extensions/CredentialProviderExtensions.cs]: was this dead code?

**Comment by onovotny** [comment]
Thanks - will review, clean up and merge this sometime tomorrow.

File: Core/Packages/ZipPackageFile.cs [hunk 4]
```diff
@@ -2,18 +2,19 @@
...
         private readonly Func<Stream> _streamFactory;
 
-        public ZipPackageFile(PackagePart part) 
-            : base(UriUtility.GetPath(part.Uri))
+        public ZipPackageFile(PackageArchiveReader reader, string path) 
+            : base(path.Replace('/', '\\'))
         {
-            Debug.Assert(part != null, "part should not be null");
-            _streamFactory = () => part.GetStream();
+            Debug.Assert(reader != null, "part should not be null");
```
Comment (304NotModified) [inline Core/Packages/ZipPackageFile.cs]: please update also the comment :)

File: Core/Extensions/CredentialProviderExtensions.cs (hunk 3 above)
Comment (onovotny) [reply Core/Extensions/CredentialProviderExtensions.cs]: seemed to be, nothing broke once it was gone :)

File: Core/Packages/ZipPackageFile.cs (hunk 4 above)
Comment (onovotny) [reply Core/Packages/ZipPackageFile.cs]: done

**Comment by onovotny** [comment]
@304NotModified I think this PR is ready. Nupkg's from local or a feed now are read through the `PackageArchiveReader` type. The original issue re schema validation should be resolved by this since we are no longer reading the XML directly.

**Comment by 304NotModified** [comment]
@onovotny the changes after the review, are those refactorings, new features or bugfixes? 

Does it need a new test?

**Comment by onovotny** [comment]
The changes were realizing that the original issue #224 wasn't actually addressed by this yet. The OM was updated, but the validation errors would occur. The subsequent changes were to actually address that issue.

**Comment by 304NotModified** [comment]
OK thanks! This is really a new major version indeed 👍 

Will try to test tomorrow!

**Comment by onovotny** [comment]
Closed since #252 contains these changes

**Pull request closed without merging**